File: threedollar-domain/src/main/java/com/depromeet/threedollar/domain/user/domain/visit/VisitHistoryCreator.java
package com.depromeet.threedollar.domain.user.domain.visit;

import com.depromeet.threedollar.common.docs.ObjectMother;
import com.depromeet.threedollar.domain.user.domain.store.Store;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@ObjectMother
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class VisitHistoryCreator {

    public static VisitHistory create(Store store, Long userId, VisitType type, LocalDate dateOfVisit) {
        return VisitHistory.builder()
            .store(store)
            .userId(userId)
            .type(type)
            .dateOfVisit(dateOfVisit)
            .build();
    }

}

File: threedollar-domain/src/testFixtures/java/com/depromeet/threedollar/domain/user/domain/visit/VisitHistoryCreator.kt
package com.depromeet.threedollar.domain.user.domain.visit

import com.depromeet.threedollar.domain.user.domain.TestFixture
import com.depromeet.threedollar.domain.user.domain.store.Store
import java.time.LocalDate

@TestFixture
object VisitHistoryCreator {

    @JvmStatic
    fun create(
        store: Store,
        userId: Long,
        type: VisitType,
        dateOfVisit: LocalDate
    ): VisitHistory {
        return VisitHistory.builder()
            .store(store)
            .userId(userId)
            .type(type)
            .dateOfVisit(dateOfVisit)
            .build()
    }

}
